src/Tribe/Views/V2/functions/classes.php

```diff
@@ -94,3 +96,64 @@ function month_multiday_classes( $event, $day_date, $is_start_of_week, $today_da
 	 */
 	return apply_filters( 'tribe_events_views_v2_month_multiday_classes', $classes, $event, $day_date, $is_start_of_week, $today_date );
 }
+
+/**
+ * Used in the Month View days loop.
+ * Outputs classes for each day "cell".
+ *
+ * @since TBD
+ *
+ * @param array<mixed> $day          The current day data.
+ * @param string       $day_date     The current day date, in the `Y-m-d` format.
+ * @param \DateTime    $request_date The request date for the view.
+ * @param string       $today_date   Today's date in the `Y-m-d` format.
+ *
+ * @return array<string,bool> $day_classes The classes to add to the day "cell".
+ */
+function month_day_classes( array $day, string $day_date, \DateTime $request_date, string $today_date ) {
+	// If for some reason we don't have a request date, use today's date.
+	$comparison_date = ! empty( $request_date ) ? $request_date->format( 'Y-m-d' ) : $today_date;
+
+	/**
+	 * Allows filtering the date used for comparison when generating the Month View day cell classes.
+	 *
+	 * @since TBD
+	 *
+	 * @param string       $comparison_date The date used for comparisons.
+	 * @param DateTime     $request_date    The request date for the view.
+	 * @param string       $day_date        The current day date, in the `Y-m-d` format.
+	 * @param array<mixed> $day             The current day data.
+	 */
+	$comparison_date =  apply_filters( 'tec_events_month_day_classes_comparison_date', $comparison_date, $request_date, $day_date, $day  );
+
+	// Convert it to a date object.
+	$comparison_date = Dates::build_date_object( $comparison_date );
```

Comment:
Use the immutable version of the object:
```suggestion
	$comparison_date = Dates::immutable( $comparison_date );
```
When the reference to a date object is passed around, as is the case in the filter following this code, then there's a chance the filtering code will modify it by reference for all the following filters.
You should always use immutable date-time objects, and use the non-mutable version only when in complete control of the flow. Too many bugs to number emerged during CT1 development due to this.